src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.StackTraceExplorer
+{
+    /// <summary>
+    /// A line of text that was parsed by <see cref="StackTraceAnalyzer" />
+    /// to provide metadata bout the line. 
+    /// </summary>
+    internal class ParsedStackFrame : ParsedFrame
+    {
+        public ParsedStackFrame(string originalText, TextSpan classSpan, TextSpan methodSpan, TextSpan argsSpan)
+            : base(originalText)
+        {
+            Contract.ThrowIfTrue(classSpan.IsEmpty);
+            Contract.ThrowIfTrue(methodSpan.IsEmpty);
+
+            ClassSpan = classSpan;
+            MethodSpan = methodSpan;
+            ArgsSpan = argsSpan;
+        }
+
+        /// <summary>
+        /// The full classname parsed from the line. 
+        /// e.x: [|Microsoft.CodeAnalysis.Editor.CallstackExplorer|].Example(arg1, arg2)
+        /// </summary>
+        public TextSpan ClassSpan { get; }
```

Comment:
'parsed' is a bit odd in the doc as it's just a TextSpan.  Furthermore, can you call this FullClassNameSpan.  'ClassSpan' is a bit vague.